mh/arena_utils.py

```diff
@@ -0,0 +1,200 @@
+            GRUDGE_MATCH += slot(*weapon3)
+        else:
+            GRUDGE_MATCH += b"\xff" * 0x04
+        # Armors
+        GRUDGE_MATCH += slot(EquipmentClasses.Helmet, helm)
+        GRUDGE_MATCH += slot(EquipmentClasses.Chestpiece, chestpiece)
+        GRUDGE_MATCH += slot(EquipmentClasses.Faulds, faulds)
+        GRUDGE_MATCH += slot(EquipmentClasses.Gauntlets, gauntlets)
+        GRUDGE_MATCH += slot(EquipmentClasses.Leggings, leggings)
+        # Items (24 4-byte slots available)
+        for item in items:
+            GRUDGE_MATCH += item_slot(*item)
+        GRUDGE_MATCH += b"\x00" * (24*4 - len(items)*4)
+        # Gunner pouch (8 4-byte slots available)
+        for item in gunner_pouch:
+            GRUDGE_MATCH += item_slot(*item)
+        GRUDGE_MATCH += b"\x00" * (8*4 - len(gunner_pouch)*4)
+
+
+GRUDGE_MATCH_ROYAL_LUDROTH = make_binary_event_quest(0xEA61, "Grudge Match: Royal Ludroth", "Announcer/Receptionist", "Slay a Royal Ludroth", generate_flags((0,0,0,0,0,1,0,0),(1,0,0,0,1,0,0,0),(0,0,0,0,0,0,0,0),(1,0,0,0,1,0,1,0)),  # quest id, name, client, description, flags
+    Monster.royal_ludroth, 0x0000, True, 0x17, 0x64, 0x00, 0x00,  # monster 1:  type, boss id, enabled, level, size, min, max
+    Monster.none, 0x0000, False, 0x00, 0x00, 0x00, 0x00,  # monster 2:  type, boss id, enabled, level, size, min, max
+    Monster.none, 0x0000, False, 0x00, 0x00, 0x00, 0x00,  # monster 3:  type, boss id, enabled, level, size, min, max
+    1000, 0, 0, 350, 0, 50, Monster.none, Monster.none,  # main reward, sub1 reward, sub2 reward, cart penalty, quest fee, time limit, mainmonst1, mainmonst2
+    LocationType.QUEST_LOCATION_WATER_ARENA_2, QuestRankType.star_1, 0, QuestRestrictionType.RESTRICTION_NONE, ResourcesType.arena, 0,  # location, quest rank, hrp reward, rank requirement, resources type, supply set number
+    0x00000101, Monster.royal_ludroth, 0x01,  # Main quest: type, target, number
+    "None.", 0x00000000, Monster.none, 0x00,  # Subquest 1: Description, type, target, number
+    "None.", 0, Monster.none, 0,  # Subquest 2: Description, type, target, number
+    "Wanted:" + '\x0A' + "The description for this" + '\x0A' +
+        "quest! If you can find" + '\x0A' + "it, please let us know!" + '\x0A' +
+        "Thanks!",  # Quest details
+    StartingPositionType.camp, 0x0017, 0x00000002,  # Start position, general enemy level, unk_12 (2 for large mon quest, 3 for small/delivery, 5 for jhen/ala)
+    0x00000000, 0x00000000, 0x00, 0x00, 0x00, 0x00000000,  # Subquest1 HRP, Subquest2 HRP, Unknown 4, Unknown 5, Unknown 6, Unknown 7
+    0x00000000, 0x00000000, 0x00000000, 0x00000000,  # Unknown 9, Unknown 10, Unknown 11, Summon
+    'sm_underwaterarenarock.dat', # Small monster data
+    [(ItemsType.r_ludroth_coin, 1, 24), (ItemsType.r_ludroth_coin, 2, 8), (ItemsType.voucher, 1, 10), (ItemsType.armor_sphere, 1, 24),
+        (ItemsType.steel_eg, 1, 18), (ItemsType.pinnacle_coin, 1, 16)],
+    [],
+    [],
+    [])
```

Comment:
Please run PEP8 or pycodestyle on all the files you're creating and address the coding style issues.